File: Java/src/main/java/com/backend/java/model/VideoContent.java
package com.backend.java.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonTypeName;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

@Entity
@Table(name = "video_content")
@JsonTypeName("video")
public class VideoContent extends Content {

    @JsonProperty("videoUrl")
    private String videoUrl;
    @JsonProperty("duration")
    private int duration;

    // Constructors, getters, and setters
    public VideoContent() {}

    public VideoContent(String videoUrl, int duration) {
        this.videoUrl = videoUrl;
        this.duration = duration;
    }

    public String getVideoUrl() {
        return videoUrl;
    }

    public void setVideoUrl(String videoUrl) {
        this.videoUrl = videoUrl;
    }

    public int getDuration() {
        return duration;
    }

    public void setDuration(int duration) {
        this.duration = duration;
    }

    @Override
    public String toString() {
        return "VideoContent{" +
                "videoUrl='" + videoUrl + '\'' +
                '}';
    }
}

File: Csharp/Models/VideoContent.cs
namespace Models
{
    public class VideoContent : Content
    {
        public string VideoUrl { get; set; }
        public int Duration { get; set; }

        public VideoContent() 
        {
            Type = "video"; 
        }

        public VideoContent(string videoUrl, int duration)
        {
            VideoUrl = videoUrl;
            Duration = duration;
            Type = "video"; 
        }
    }
}